### PR TITLE
Update time.py

### PR DESCRIPTION
**Pull request by ankitkmr**
This offers a solution to the issue #798

### CODE REVIEW

**Comment by ankitkmr** [comment]
@dpshelio Is my fix incorrect?

File: sunpy/time/time.py [hunk 1]
```diff
@@ -188,8 +190,21 @@ def parse_time(time_string, time_format=''):
...
+        if '+' in time_string :
+            time_zone = time_string[time_string.rindex('+'):]
+            time_string_strip = time_string[:time_string.rindex('+')].rstrip("0").rstrip(".")
+            d = parse(time_string_strip + time_zone)
+            return d.astimezone(timezone('UTC'))
+        elif '-' in time_string :
+            time_zone = time_string[time_string.rindex('-'):]
+            time_string_strip = time_string[:time_string.rindex('-')].rstrip("0").rstrip(".")
+            d = parse(time_string_strip + time_zone)
+            return d.astimezone(timezone('UTC'))
```
Comment (dpshelio) [inline sunpy/time/time.py]: All this probably could be done using just one `if` and a regular expression like:
```python
time_zone = time_string[re.search('\+|\-', time_string).start():]
```
so we can avoid the code repetition.

**Comment by dpshelio** [comment]
Thanks @ankitkmr for your contribution.
It would be good if you could add a test to cover these cases, so we know it works properly.

**Comment by dpshelio** [comment]
Hum... it seems that this have a lot of consequences, probably because there is time formats that contains a `-` in it, eg, [ISO8601](http://en.wikipedia.org/wiki/ISO_8601)
Maybe that's what Travis is complaining about...

**Comment by ankitkmr** [comment]
ok so I'll add test cases then .. Thanks

**Comment by ankitkmr** [comment]
yeah ok found a mistake you are right about - thing updating after checking for a few more test cases now

**Pull request closed without merging**